rbc/targetinfo.py

```diff
@@ -305,6 +306,27 @@ def has_cpython(self):
 
     # TODO: info may also contain: count, threads, cores
 
+    def supports_devices(self, desired_devices):
+        """Check if target supports desired devices
+        """
+        if desired_devices is None:
+            return True
+        for d in desired_devices:
+            d = d.lower()
+            if d == 'cpu':
+                if self.is_cpu:
+                    return True
+            elif d == 'gpu':
+                if self.is_gpu:
+                    return True
+            elif d == 'cuda':
+                if self.driver[0] == 'CUDA':
+                    return True
+            else:
+                warnings.warn(
+                    f'Ignoring unknown desired device specification: {d}')
+        return False
+
```

Comment:
`supports_devices` needs to check if every device in `desired_devices` is supported? or if at least one device is supported? If the former is the case, then, I guess this function is wrong. For instance, consider the following case: `self.supports_device(['cpu', 'gpu', 'a random string'])`. Even though target may support `cpu` and `gpu`, `a random string` is unknown.

Reply:
It's the latter. Notice that the `TargetInfo` instance `self` represents a single target device, cpu or gpu. The `self.supports_devices` method tests if the `self` target device is in the `desired_devices` list.

So, the doc string is wrong and the naming of the method could be better. Here's a plan:
```
def contained_in(self, desired_devices):
    """Checks if target is in a set of desired devices.

    Parameters
    ----------
    desired_devices: list, None
      Devices names that can be a subset of `['cpu', 'gpu', 'cuda']`.

    Returns
    -------
    is_contained : bool
      True, if the target device name is listed in `desired_devices`
      or `desired_devices` is None. False, otherwise.
    """
```
Any suggestions?

Reply:
`contained_in` seems a good name. Or maybe `supports_any_device`

Reply:
Moved this to https://github.com/xnd-project/rbc/pull/228/files